tests/zfs-tests/tests/functional/direct/dio_write_verify.ksh

```diff
@@ -30,34 +30,36 @@
 
 #
 # DESCRIPTION:
-# 	Verify checksum verify works for Direct IO writes.
+# 	Verify checksum verify works for Direct I/O writes.
 #
 # STRATEGY:
-#	1. Set the module parameter zio_direct_write_verify to 1.
-#	2. Start FIO Direct IO write workload and first make sure we do not
-#          have any failures using zpool status -d or zevent.
-#	3. Start a Direct IO write workload while manipulating the user buffer.
-#	4. Verify there are Direct IO write verify failures using
-#          zpool status -d and checking for zevents.
-#	5. Make sure there are no reported data errors because enabling
-#          Direct IO write verifies will not lead to any data corruption if
-#          there was checksum failure before the write was committed to a VDEV.
+#	1. Set the module parameter zfs_vdev_direct_write_verify_cnt to 10.
+#	2. Start a Direct IO write workload while manipulating the user buffer
+#          with and without compression.
+#	3. Verify there are Direct I/O write verify failures using
+#          zpool status -d and checking for zevents. We also make sure there
+#          are reported data errors when reading the file back.
+#	4. Repeat 2 but with zfs_vdev_direct_write_verify_cnt set to 1.
+#	5. Verify there are Direct I/O write verify failures using
+#          zpool status -d and checking for zevents. We also make sure there
+#          here are no reported data errors when reading the file back because
+#          with us checking every Direct I/O write and on checksum validation
+#          failure those writes will not be committed to a VDEV.
 #
 
 verify_runnable "global"
 
 function cleanup
 {
-	log_must rm -f "$mntpnt/direct-write.iso"
 	# Clearing out DIO VERIFY counts for Zpool
 	log_must zpool clear $TESTPOOL
 	# Clearing out dio_verify from event logs
 	log_must zpool events -c
-	log_must eval "echo 0 > \
-	    /sys/module/zfs/parameters/zio_direct_write_verify"
+	log_must eval "echo 100 > \
+	    /sys/module/zfs/parameters/zfs_vdev_direct_write_verify_cnt"
```

Comment:
Please add `zfs_vdev_direct_write_verify_cnt` to tunables.cfg and use `set_tunable32` to set

Reply:
Good catch. I went ahead and updated this.